pkg/services/ssosettings/api/api_test.go

```diff
@@ -253,6 +254,119 @@ func TestSSOSettingsAPI_Delete(t *testing.T) {
 	}
 }
 
+func TestSSOSettingsAPI_GetForProvider(t *testing.T) {
+	type TestCase struct {
+		desc                string
+		key                 string
+		action              string
+		scope               string
+		expectedResult      *models.SSOSettings
+		expectedError       error
+		expectedServiceCall bool
+		expectedStatusCode  int
+	}
+
+	tests := []TestCase{
+		{
+			desc:   "successfully gets SSO settings",
+			key:    "azuread",
+			action: "settings:read",
+			scope:  "settings:auth.azuread:*",
+			expectedResult: &models.SSOSettings{
+				ID:        "1",
+				Provider:  "azuread",
+				Settings:  make(map[string]interface{}),
+				Created:   time.Now(),
+				Updated:   time.Now(),
+				IsDeleted: false,
+				Source:    models.DB,
+			},
+			expectedError:       nil,
+			expectedServiceCall: true,
+			expectedStatusCode:  http.StatusOK,
+		},
+		{
+			desc:                "fails when action doesn't match",
+			key:                 "azuread",
+			action:              "",
```

Comment:
I'd go with something similar to `settings:read` here instead of empty action. Maybe `settings:write` (I haven't checked that it fails though)?